File: e2e/playwright-typescript/support/api/ApiClient.ts
import { APIRequestContext } from '@playwright/test';
import { TaskRequest } from '@/data/TaskRequest';
import { testConfig } from '@/test.config';

type TaskResponse = {
  id: string;
  title: string;
  description: string;
  status: string;
  priority: string;
};

export class ApiClient {
  private readonly baseUrl = testConfig.services.api.url;

  constructor(private readonly request: APIRequestContext) {}

  async createTask(task: TaskRequest): Promise<TaskResponse> {
    const response = await this.request.post(`${this.baseUrl}/tasks`, {
      data: task,
    });
    return response.json();
  }

  async deleteTask(id: string): Promise<void> {
    await this.request.delete(`${this.baseUrl}/tasks/${id}`);
  }

  async getTasks(): Promise<TaskResponse[]> {
    const response = await this.request.get(`${this.baseUrl}/tasks`);
    return response.json();
  }
}

File: e2e/playwright-python/support/api/api_client.py
import requests
import config
from data.task_data import TaskData
from data.task_response import TaskResponse


class ApiClient:
    def __init__(self):
        self._base_url = f"{config.api_url}/v1"

    def create_task(self, task: TaskData) -> TaskResponse:
        response = requests.post(
            f"{self._base_url}/tasks",
            json={
                "title": task.title,
                "description": task.description,
                "status": task.status,
                "priority": task.priority,
            },
        )
        response.raise_for_status()
        data = response.json()
        return TaskResponse(**data)

    def delete_task(self, task_id: str) -> None:
        response = requests.delete(f"{self._base_url}/tasks/{task_id}")
        response.raise_for_status()

    def get_tasks(self) -> list[TaskResponse]:
        response = requests.get(f"{self._base_url}/tasks")
        response.raise_for_status()
        return [TaskResponse(**item) for item in response.json()]
